src/sortStudents.ts

```diff
@@ -1,16 +1,128 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const resArr: Student[] = [...students];
+  const sortOrder: boolean = order === 'asc';
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  switch (sortBy) {
+    case SortType.Name:
+      resArr.sort((a: Student, b: Student): number => {
+        if (sortOrder) {
+          return a.name.localeCompare(b.name);
+        }
+
+        return b.name.localeCompare(a.name);
+      });
+      break;
+    case SortType.Surname:
+      resArr.sort((a: Student, b: Student): number => {
+        if (sortOrder) {
+          return a.surname.localeCompare(b.surname);
+        }
+
+        return b.surname.localeCompare(a.surname);
+      });
```

Comment:
You could write it with one case because the code is nearly the same.

And use ternary operator instead of `if` all over the code.
```suggestion
    case SortType.Name:
    case SortType.Surname:
    .....
```